lib/elixir/lib/macro.ex

```diff
@@ -8,14 +8,43 @@ defmodule Macro do
 
   To create a custom sigil, define a function with the name
   `sigil_{identifier}` that takes two arguments. The first argument will be
-  the interpolated string, the second will be a char list containing any
-  modifiers.
+  the string, the second will be a char list containing any modifiers. If the
+  sigil is lower case (such as `sigil_x`) then the string argument will allow
+  interpolation. If the sigil is upper case (such as `sigil_X`) then the string
+  will not be interpolated.
 
   Valid modifiers include only lower and upper case letters. Other characters
   will cause a syntax error.
 
   The module containing the custom sigil must be imported before the sigil
   syntax can be used.
+
+  ### Examples
+
+      defmodule MySigils do
+        defmacro sigil_x(term, [?r]) do
+          quote do: unquote(term) |> String.reverse()
+        end
+        defmacro sigil_x(term, _modifiers) do
+          term
+        end
+        defmacro sigil_X(term, [?r]) do
+          quote do: unquote(term) |> String.reverse()
+        end
+        defmacro sigil_X(term, _modifiers) do
+          term
+        end
+      end
+
+      iex > import MySigils
+      iex > ~x(with #{"inter" <> "polation"})
+      "with interpolation"
+      iex > ~x(with #{"inter" <> "polation"})r
+      "noitalopretni htiw"
+      iex > ~X(without #{"interpolation"})
+      "without \#{"interpolation"}"
+      iex > ~X(without #{"interpolation"})r
```

Comment:
> Just to clarify, I have added a space in each `iex >` in the examples so that the build does not fail.

Is there a way to define to module (`MySigils`) in a way that it can be used in a doctest?